dist/jose-jwe.js

```diff
@@ -658,18 +667,18 @@ JoseJWE.prototype.decrypt = function(key_promise, cipher_text) {
   }
 
   // part 1: header
-  header = JSON.parse(Utils.Base64Url.decode(parts[0]));
-  if (!header.alg) {
+  this.headers = JSON.parse(Utils.Base64Url.decode(parts[0]));
```

Comment:
Keep managed header parameters out of the user headers. You don't want to imply that setting `alg` or `enc` through `addHeader` has any effect.